app/src/main/java/com/github/sdpteam15/polyevents/helper/GoogleMapHelper.kt

```diff
@@ -0,0 +1,429 @@
+                )
+            ).anchor(0.5f, 0.5f).draggable(true).snippet(PolygonAction.RIGHT.toString())
+        )
+        moveRightPos = moveRightMarker!!.position
+        moveDownMarker = map!!.addMarker(
+            MarkerOptions().position(posMidDown).icon(
+                getMarkerRessource(
+                    R.drawable.ic_downarrow
+                )
+            ).anchor(0.5f, 0.5f).draggable(true).snippet(PolygonAction.DOWN.toString())
+        )
+        moveDownPos = moveDownMarker!!.position
+        moveMarker = map!!.addMarker(
+            MarkerOptions().position(posCenter).icon(getMarkerRessource(R.drawable.ic_move))
+                .anchor(0.5f, 0.5f).draggable(true).snippet(
+                PolygonAction.MOVE.toString()
+            )
+        )
+        movePos = moveMarker!!.position
+    }
+
+    /**
+     * Generates the icon for the invisible icons
+     * ref : https://stackoverflow.com/questions/35718103/how-to-specify-the-size-of-the-icon-on-the-marker-in-google-maps-v2-android
+     * TODO : Check if we should make it a singleton to save memory/performance
+     */
+    private fun getMarkerRessource(id: Int): BitmapDescriptor {
+        val vectorDrawable: Drawable? = ContextCompat.getDrawable(context!!, id)
+        vectorDrawable?.setBounds(0, 0, 100, 100)
+        val bitmap = Bitmap.createBitmap(100, 100, Bitmap.Config.ARGB_8888)
+        val canvas = Canvas(bitmap)
+        vectorDrawable?.draw(canvas)
+        return BitmapDescriptorFactory.fromBitmap(bitmap)
+    }
+
+    /**
+     * Translate all the corners and the edition markers from the rectangle by the same distance the "moveMarker" is moved
+     */
+    fun translatePolygon(pos: Marker) {
+        val diffLat = pos.position.latitude - movePos!!.latitude
+        val diffLng = pos.position.longitude - movePos!!.longitude
+        tempLatLng = tempLatLng.map { latLng ->
+            LatLng(
+                latLng!!.latitude + diffLat,
+                latLng.longitude + diffLng
+            )
+        }.toMutableList()
+
+        //Moves the edition markers
+        moveMarker!!.position = LatLng(movePos!!.latitude + diffLat, movePos!!.longitude + diffLng)
+        movePos = moveMarker!!.position
+
+        moveDiagMarker!!.position =
+            LatLng(moveDiagPos!!.latitude + diffLat, moveDiagPos!!.longitude + diffLng)
+        moveDiagPos = moveDiagMarker!!.position
+
+        moveRightMarker!!.position =
+            LatLng(moveRightPos!!.latitude + diffLat, moveRightPos!!.longitude + diffLng)
+        moveRightPos = moveRightMarker!!.position
+
+        moveDownMarker!!.position =
+            LatLng(moveDownPos!!.latitude + diffLat, moveDownPos!!.longitude + diffLng)
+        moveDownPos = moveDownMarker!!.position
+    }
+
+    /**
+     * Transforms the size of the rectangle, either by moving the the right wall(right), the down wall(down) or both(diag)
+     */
+    fun transformPolygon(pos: Marker) {
```

Comment:
It would be nice to introduce a bit more modularity and split this method into smaller methods.